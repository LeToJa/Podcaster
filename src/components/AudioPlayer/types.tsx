export interface AudioPlayerTypes {
    url: string
    type: string
}