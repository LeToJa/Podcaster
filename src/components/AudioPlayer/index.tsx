import { AudioPlayerTypes } from "./types"

const AudioPlayer = ({ url, type }: AudioPlayerTypes) => <>
    <audio controls>
        <source src={url} type={type} />
        Your browser does not support the audio element.
    </audio>
</>

export default AudioPlayer