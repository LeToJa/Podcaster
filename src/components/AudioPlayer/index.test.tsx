import { render } from '@testing-library/react'
import { describe, it, expect } from 'vitest'

import AudioPlayer from '.'

const mockProps = {
    url: 'http://example.com/audio.mp3',
    type: 'audio/mpeg',
}

describe('AudioPlayer', () => {
    it('renders correctly and matches snapshot', () => {
        const { container } = render(<AudioPlayer {...mockProps} />)

        expect(container).toMatchSnapshot()
    })

    it('renders correctly with given props', () => {
        const { container, getByText } = render(<AudioPlayer {...mockProps} />)

        const sourceElement = container.querySelector('source')

        expect(container.querySelector('audio')).toBeTruthy()
        expect(sourceElement).toBeTruthy()
        expect(sourceElement).toHaveProperty('src', mockProps.url)
        expect(sourceElement).toHaveProperty('type', mockProps.type)
        expect(getByText(/Your browser does not support the audio element./i)).toBeTruthy()
    })
})
