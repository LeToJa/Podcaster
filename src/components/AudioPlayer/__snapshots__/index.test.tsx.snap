// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`AudioPlayer > renders correctly and matches snapshot 1`] = `
<div>
  <audio
    controls=""
  >
    <source
      src="http://example.com/audio.mp3"
      type="audio/mpeg"
    />
    Your browser does not support the audio element.
  </audio>
</div>
`;
